src/uct/sm/scopy/cma/cma_ep.c

```diff
@@ -125,7 +123,10 @@ ucs_status_t uct_cma_ep_put_zcopy(uct_ep_h tl_ep, const uct_iov_t *iov, size_t i
                                   uint64_t remote_addr, uct_rkey_t rkey,
                                   uct_completion_t *comp)
 {
-    UCT_CHECK_IOV_SIZE(iovcnt, uct_sm_get_max_iov(), "uct_cma_ep_put_zcopy");
+    UCT_CHECK_IOV_SIZE(iovcnt,
+                       ucs_derived_of(tl_ep->iface,
+                                      uct_scopy_iface_t)->config.max_iov,
```

Comment:
can we have local var cma_iface?

Reply:
maybe better to move it inside `uct_cma_ep_common_zcopy` and pass `"uct_cma_ep_put/get_zcopy"` depending on the function cb passed - i.e. `process_vm_writev`/`process_vm_writev`?

Reply:
yes, if it will fit your design of scopy

Reply:
> yes, if it will fit your design of scopy

it will be moved to SCOPY layer then
better to not do it then for simplicity reason

done